src/Engine/Combat/CombatContext.java

```diff
@@ -37,6 +37,12 @@ public boolean isSim()
   public CombatContext(GameInstance gi, GameMap map,
                         UnitContext pAttacker, UnitContext pDefender,
                         CalcType pCalcType)
+  {
+    this(gi, map, pAttacker, pDefender, calcBattleRange(pAttacker, pDefender), pCalcType);
+  }
+  public CombatContext(GameInstance gi, GameMap map,
+                        UnitContext pAttacker, UnitContext pDefender, int battleRange,
+                        CalcType pCalcType)
```

Comment:
I'm not sure what problem this addresses, I'm just thrilled that I'm that much closer to being able to implement teleporting ninjas. You can attack them from far away, and they will _still_ stab you in the face.

Reply:
I'm sure I could dig up the usecase, but I haven't stared at Wally code in a few weeks now.

Reply:
No idea. Haven't tried. Probably so.